base/applications/charmap/precomp.h

```diff
@@ -19,10 +19,17 @@
 #define XLARGE 45
 #define YLARGE 25
 
-#define FM_SETFONT  (WM_USER + 1)
-#define FM_GETCHAR  (WM_USER + 2)
-#define FM_SETCHAR  (WM_USER + 3)
-#define FM_GETHFONT (WM_USER + 4)
+#define FM_SETFONT    (WM_USER + 1)
+#define FM_GETCHAR    (WM_USER + 2)
+#define FM_SETCHAR    (WM_USER + 3)
+#define FM_GETHFONT   (WM_USER + 4)
+#define FM_SETCHARMAP (WM_USER + 5)
+
+// the code pages to display in the advanced 'character set' combobox
+static const UINT codePages[] = { 
```

Comment:
You can potentially inspire yourself by list of character sets (!= code pages) as done in
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/comdlg32/fontdlg.c;h=ecb6ce0e33abec48841873c54cd0dec9b65dc77d;hb=HEAD

Reply:
> 
> 
> There's an interesting article here which has some insight into ways in which it can be improved upon : https://www.osnews.com/story/28632/the-peculiar-history-of-the-windows-character-map/
> 
> I seem to remember I started a `charmap_new` a long while back as a bit of fun but never got around to finishing it. It could be a good base to use if you decided to go for a more modern design. I think I'd based it on the more modern Mac app

I was wondering where charmap_new came from!  Good to know the history behind it.  I'll take a look after I finish up my list for the existing charmap.